lib/sqlalchemy/orm/properties.py

```diff
@@ -335,3 +358,14 @@ def reverse_operate(self, op, other, **kwargs):
 
     def __str__(self):
         return str(self.parent.class_.__name__) + "." + self.key
+
+
+class _ColumnPropertyProxy(ColumnProperty):
+    def __init__(self, *columns, **kwargs):
+        # will be changed to True in a future release
+        kwargs.setdefault("raise_on_set", "warn")
+        super().__init__(*columns, **kwargs)
+
+
+_ColumnPropertyProxy.__doc__ = ColumnProperty.__doc__
+_ColumnPropertyProxy.__init__.__doc__ = ColumnProperty.__init__.__doc__
```

Comment:
I'm not particularly happy by this implementation to change the default value in the `column_property` while keeping the default to False in the `ColumnProperty`.

I think that `public_factory` should handle this. What do you think?